examples/dev_example/docs/build.jl

```diff
@@ -0,0 +1,28 @@
+import Lexicon.Elements: config, section, document, page
+
+import Lexicon.Render: save, markdown
+
+import Docile: Cache, Collector
+
+out = document("TestDocumentation",
+    section("Introduction",
+        page("index", "index.md"),
+        ),
+    section("Docile-API",
+        page("Cache", config(style_obj="###"),
+            "# Cache",
+            "## Macros",
+            "Some intermediate text describing more about macros.",
+            (Cache, config(objfilter=[:macro])),
+            "## Methods",
+            (Cache, config(objfilter=[:method])),
+            ),
+        page("Collector", config(objfilter=[:method, :macro], style_obj="###"),
+            "# Collector",
+            "## Methods & Macros",
+            Collector,
+            ),
+        ),
+    )
```

Comment:
> Could we use a Symbol for the name then?

I anyway do your suggested `NodeN{}`  I will implement it like this: 

`typealias NodeName Union(Symbol, AbstractString)`